### DISCUSSION TITLE
Category toggle buttons are reset to false when Preference modal is closed

### DISCUSSION THREAD

**Discussion by matyexd** [orestbida/cookieconsent#668] [Q&A] [answered]
React 18.2.0, vanilla-cookieconsent v3

Set category parameters
```javascript
      categories: {
        necessary: {
          enabled: true,
          readOnly: true,
        },
        functionality: {
          enabled: true,
        },
        analytics: {
          enabled: true,
        },
        marketing: {
          enabled: true,
        },
        personalization: {
          enabled: true,
        },
        security: {
          enabled: true,
        },
      },
``` 
But after closing the Preference modal window, the toggle buttons to false

https://github.com/orestbida/cookieconsent/assets/72869563/5536cd36-4f8d-43da-a772-bec72984bef9

**Comment by orestbida** [comment, marked as answer]
@matyexd can you try the [latest ](https://github.com/orestbida/cookieconsent/blob/master/dist/cookieconsent.esm.js)version in the master branch? should be fixed.

**Reply by matyexd** [reply]
Thank you very much!

**Reply by orestbida** [reply]
@matyexd, released 3.0.1 with the fix.
